### ISSUE TITLE
Store additional information in TLorentzVector

### ISSUE THREAD

**Issue by dguest** [scikit-hep/uproot-methods#40]
I [asked earlier][0] about constructing the Lorentz vector objects directly from numpy arrays. That works really well! But I'd also like to be able to add additional information to the objects, i.e. in some cases we have other jet characteristics like b-tagging scores. Is there some method to append additional information (which is stored in another array) to each object?

[0]: https://github.com/scikit-hep/awkward-array/issues/9

**Comment by jpivarski** [comment]
Yes. A TLorentzVectorArray contains a Table, so you should be able to attach or remove any other arrays to it, as long as they fit (same number of elements, and if jagged, same number in each subarray).

```python
a = TLorentzVectorArray.from_ptetaphim(pt, eta, phi, mass)
a["isolation"] = isolation
```

and so on.

**Comment by dguest** [comment]
OK, that's ridiculously easy.

**Comment by dguest** [comment]
OK, while this obviously isn't the recommended way to do things, I noticed that when I looped over the  jets the iterator seemed to return a regular TLorentzVector without the isolation attached. Is there some way to get both together? I noticed that the following seems to work:

```
for vec, iso in zip(vecarray, vecarray.content['isolation']):
    print(vec, iso)
```

but it's a bit clunky.

**Comment by jpivarski** [comment]
To get that to work, I should have made the TLorentzVector object (singular) a mixin of Table.Row, instead of a separate class. At some point, I should do that so that you can get isolation from `vec["isolation"]`.

I'll reopen this to remind myself to go back and do that.
